packages/headless/src/app/search-engine/jwt-reducer.ts

```diff
@@ -0,0 +1,223 @@
+import {AnyAction, Reducer} from '@reduxjs/toolkit';
+import {SearchAppState} from '../..';
+import {atob as atobshim} from 'abab';
+import {isNullOrUndefined} from '@coveo/bueno';
+import P, {Logger} from 'pino';
+import {setSearchHub} from '../../features/search-hub/search-hub-actions';
+import {
+  updateAnalyticsConfiguration,
+  updateSearchConfiguration,
+} from '../../features/configuration/configuration-actions';
+import {getSearchHubInitialState} from '../../features/search-hub/search-hub-state';
+import {setPipeline} from '../../features/pipeline/pipeline-actions';
+import {getPipelineInitialState} from '../../features/pipeline/pipeline-state';
+import {getConfigurationInitialState} from '../../features/configuration/configuration-state';
+
+export interface CoveoJSONWebToken {
+  searchHub?: string;
+  pipeline?: string;
+  userDisplayName?: string;
+}
+
+const possiblyWarnOnMismatch = (
+  token: CoveoJSONWebToken,
+  tokenProp: keyof CoveoJSONWebToken,
+  stateProp: string,
+  defaultProp: string,
+  payload: string | undefined,
+  logger: Logger
+) => {
+  const tokenValue = token[tokenProp];
+  if (isNullOrUndefined(tokenValue)) {
+    return;
+  }
+
+  if (stateProp === tokenValue) {
+    return;
+  }
+
+  if (isNullOrUndefined(payload)) {
+    return;
+  }
+
+  if (payload === tokenValue) {
+    return;
+  }
+
+  if (payload === defaultProp) {
+    return;
+  }
+
+  logger.warn(
+    `Mismatch on access token (JWT Token) ${tokenProp} and engine configuration.`
+  );
+  logger.warn(
+    `To remove this warning, make sure that access token value [${tokenValue}] matches engine configuration value [${stateProp}]`
+  );
+};
+
+const shouldReconcileValues = (
+  token: CoveoJSONWebToken,
+  tokenProp: keyof CoveoJSONWebToken,
+  stateValue: string
+): boolean => {
+  const tokenValue = token[tokenProp];
+
+  if (isNullOrUndefined(tokenValue)) {
+    return false;
+  }
+
+  if (stateValue === tokenValue) {
+    return false;
+  }
+
+  return true;
+};
+
+const decodeJSONWebToken = (token: string): CoveoJSONWebToken | false => {
+  try {
+    const atobImplementation = typeof atob !== 'undefined' ? atob : atobshim;
```

Comment:
`atob` does not exists in Node environment, you have to use Buffer module.

So instead I just added a new package to shim the implementation when it's needed.